app.js

```diff
@@ -42,8 +43,12 @@ app.use(
 );
 
 //TODO: Pass login info in main handler bar, or request log
-app.use((req, res, next) => {
+app.use(async (req, res, next) => {
```

Comment:
Is this middleware working? If so, this will be a better approach than passing user info from the router. I saw you also make the change to different routers. I believe the field you want is the user name. If so, the login router can add:
req.session.userName field with the username. And the username can be fetched from this middleware function and remove the fields passing from routers.